src/lightning_lite/plugins/precision/deepspeed.py

```diff
@@ -53,18 +56,23 @@ def __init__(self, precision: Union[str, int], amp_type: str, amp_level: Optiona
 
             amp_level = amp_level or "O2"
 
-        supported_precision = (PrecisionType.HALF, PrecisionType.FLOAT, PrecisionType.BFLOAT)
-        if precision not in supported_precision:
+        supported_precision = ("16", "32", "bf16")
+        if str(precision) not in supported_precision:
             raise ValueError(
                 f"`precision={precision!r})` is not supported in DeepSpeed."
-                f" `precision` must be one of: {(x.value for x in supported_precision)}."
+                f" `precision` must be one of: {', '.join(supported_precision)}."
             )
 
         super().__init__()
         self.precision = precision
```

Comment:
If you stop using the enums to check. Then IMO this should be saved as a string to avoid potential issues with checking `str == int`

`self.precision = str(self.precision)`

And in the dictionaries: `{"bf16": torch.bfloat16, "16": torch.float16, "32": torch.float32}`

Reply:
Yes, I had the impression we decided to move away from the enums a while ago, although I don't remember all the arguments. 

I agree it would be good to only have the string type. I need to check where the precision attribute is used, connector etc.

Reply:
See #12069 which is related

Reply:
@carmocca @awaelchli while on it and while adding a new package, should we also integrate https://github.com/Lightning-AI/lightning/issues/9956#issuecomment-1207246337 to avoid the deprecation cycle here?

Reply:
> I had the impression we decided to move away from the enums a while ago, although I don't remember all the arguments.

The argument is that they are not easy to extend.

Let's continue the discussion about planning for #9956 offline.

> I agree it would be good to only have the string type. I need to check where the precision attribute is used, connector etc.

From this comment I assume you still need to set it as a str